osu.Game/Online/Multiplayer/MultiplayerRoomSettings.cs

```diff
@@ -32,13 +32,22 @@ public class MultiplayerRoomSettings : IEquatable<MultiplayerRoomSettings>
         [Key(4)]
         public IEnumerable<APIMod> Mods { get; set; } = Enumerable.Empty<APIMod>();
 
+        [NotNull]
+        [Key(5)]
+        public IEnumerable<APIMod> AllowedMods { get; set; } = Enumerable.Empty<APIMod>();
```

Comment:
Shall we call this `OptionalMods`? Allowed seems a bit weird since it doesn't include the Mods above.

Reply:
I would earlier change the above to `RequiredMods` to match the osu-web API.

Reply:
Sounds fine to do that?

Reply:
It would break iOS though, due to JSON serialization by name. It shouldn't break messapack though, right?

Reply:
ios hasn't even had a release supporting multiplayer, not important at this stage.

Reply:
Sure. Do you want that done in this PR?

Reply:
Yeah. They can't exist with the current naming pair; feels super weird.